test/new-e2e/pkg/environments/aws/host/host.go

```diff
@@ -118,6 +121,14 @@ func WithoutAgent() ProvisionerOption {
 	}
 }
 
+// WithDocker installs docker on the VM
+func WithDocker() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.installDocker = true
+		return nil
+	}
+}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Having docker in an host environment prevents from having the docker client on the output environment struct. I'd rather use the `awsdocker` environment that has a docker client. The provider there takes care of defaulting to AMIs that have docker already installed, which saves some setup time. But we need a way to allow installing the agent on docker or on the host, so maybe it makes sense to use `awshost` for this goal. 

One way to address the issue of extending default providers is to export the run function, so that you can call it from your custom provider with a custom env based on `awshost`. Would that work for you ?